libs/zglfw/src/zglfw.zig

```diff
@@ -596,6 +596,19 @@ pub const Window = opaque {
     }
     extern fn glfwSetWindowTitle(window: *Window, title: [*:0]const u8) void;
 
+    pub fn getClipboardString(window: *Window) ?[:0]const u8 {
+        return std.mem.span(glfwGetClipboardString(window));
```

Comment:
```suggestion
        return if (glfwGetClipboardString(window)) |str| std.mem.span(str) else null;
```

Reply:
not gonna pretend i knew this before lol but it actually does work.
```zig
pub fn span(ptr: anytype) Span(@TypeOf(ptr)) {
    if (@typeInfo(@TypeOf(ptr)) == .Optional) {
        if (ptr) |non_null| {
            return span(non_null);
        } else {
            return null;
        }
...
```
from the stdlib

Reply:
Oh, nice!